Mac Os Warning: implicit declaration of function 'pthread_setname_np'

Compiling thpool on Mac Os with lldb 370.0.42 triggers the above message.

```
cc -Wall -D_GNU_SOURCE -std=gnu99 -O2 -g  -c -o dep/thpool/thpool.o dep/thpool/thpool.c
dep/thpool/thpool.c:298:3: warning: implicit declaration of function 'pthread_setname_np' is invalid in C99
      [-Wimplicit-function-declaration]
  pthread_setname_np(thread_name);
  ^
```

Removing 
```c
#define _POSIX_C_SOURCE 200809L
```
which was added in a3916021c837e4892667e02e1681437ddc362725 gets rid of the warning, and passes on Linux as well.

Any hint on how to fix this warning? Is it safe to just comment out the _POSIX_C_SOURCE line?

--- a/thpool.c
+++ b/thpool.c
@@ -8,7 +8,13 @@
  *
  ********************************/
 
+#if defined(__APPLE__)
+#include <AvailabilityMacros.h>
+#else
+#ifndef _POSIX_C_SOURCE
 #define _POSIX_C_SOURCE 200809L
+#endif
+#endif
 #include <unistd.h>
 #include <signal.h>
 #include <stdio.h>

